DOC-10406 Clarify debug zip --timeout flag as "per request" (#18613)
cockroachdb/docs: src/current/v24.1/cockroach-debug-zip.md

`--nodes` | Specify nodes to inspect as a comma-separated list or range of node IDs. For example:<br /><br />`--nodes=1,10,13-15`
`--redact` | Redact sensitive data from the generated `.zip`, with the exception of range keys, which must remain unredacted because they are essential to support CockroachDB. This flag replaces the deprecated `--redact-logs` flag, which only applied to log messages contained within `.zip`. See [Redact sensitive information](#redact-sensitive-information) for an example.
`--redact-logs` | **Deprecated** Redact sensitive data from collected log files only. Use the `--redact` flag instead, which redacts sensitive data across the entire generated `.zip` as well as the collected log files. Passing the `--redact-logs` flag will be interpreted as the `--redact` flag.

> `--timeout` | Return an error if the command does not conclude within a specified nonzero value. The timeout is suffixed with `s` (seconds), `m` (minutes), or `h` (hours). In the process of generating a debug zip, many requests are made. Each request is subject to this timeout limit.<br /><br />**Default:** `60s`

Reviewer: Optional suggestions for readability (not sure if "internal" is correct):

```suggestion
`--timeout` | Return an error if the `debug zip` command does not conclude within a specified nonzero value. The timeout is suffixed with `s` (seconds), `m` (minutes), or `h` (hours).<br /><br />In the process of generating a debug zip, many internal requests are made. Each request is subject to this timeout limit.<br /><br />**Default:** `60s`
```

Author: discussed with @taroface and updated.

> `--timeout` | In the process of generating a debug zip, many internal requests are made. Each request is subject to this timeout limit.<br /><br />Return an error if an internal request does not conclude within a specified nonzero value. The timeout is suffixed with `s` (seconds), `m` (minutes), or `h` (hours).<br /><br />**Default:** `60s`

Reviewer: This seems more straightforward to me, but I'm not sure how much better it is. I'll defer to you.

```suggestion
`--timeout` | Return an error if any of the multiple requests used to generate a debug zip does not conclude within a specified nonzero value. The timeout is suffixed with `s` (seconds), `m` (minutes), or `h` (hours).<br /><br />**Default:** `60s`
```
